### ISSUE TITLE
uriSegment in Content Dimensions is not allowed to contain "_" (underscore)

### ISSUE THREAD

**Issue by skurfuerst** [neos/neos-development-collection#2417]
### Description

The `uriSegment` property of a dimension preset must not contain an underscore. In case it does, e.g. by using `de_DE` as URI segment, the dimension is not accessible in backend and frontend.

### Root Cause

The problem is that the `FrontendNodeRoutePartHandler` uses the `_` character to distinguish between *different* dimensions; so it should be forbidden to use the `_` character in a *single* dimension.

### Steps to Reproduce

1. in the Demo site, rename the `uriSegment` from `de` to `de_DE` 
2. watch the FE and BE break once you switch to german.

#### Expected behavior

I'd suggest we throw an exception saying that the URI Segment should not contain underscores.

#### Actual behavior

A weird error about routes not matching - not understandable for users :) 

### Affected Versions

All Neos versions.

**Comment by kitsunet** [comment]
it might be nice to make the separator character configurable, that woudl be a way out if you absolutely want to have this (some SEO experts are very adamant about the way those locales need to be written, sadly both underscore and dash are requested often). By making it configurable we would help this problem and solve the problem to get dashes with country + language dimension.

**Comment by skurfuerst** [comment]
@kitsunet yeah - totally agreed. For now I just added a big fat warning to https://docs.neos.io/cms/manual/content-repository/multiple-languages#configuring-a-bilingual-site (scroll a bit down) to keep others from stumbling over it; but ofc. having it configurable would be better :)
